convert/src/main/kotlin/com/github/zeldigas/text2confl/convert/confluence/ReferenceProvider.kt

```diff
@@ -39,24 +44,41 @@ class ReferenceProviderImpl(private val basePath: Path, documents: Map<Path, Pag
     ReferenceProvider {
 
     companion object {
-        private val URI_DETECTOR = "^[a-zA-Z][a-zA-Z0-9.+-]+:/{0,2}".toRegex()
+        private const val URI_DETECTOR = "^(https?|ftp)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;'()]*[-a-zA-Z0-9+&@#/%=~_|]"
+        private const val MAILTO_DETECTOR = "mailto:"
+        private const val LOCALHOST_DETECTOR = "localhost:"
+
+        private val logger = KotlinLogging.logger {}
+    }
+    fun isValid(url: String): Boolean {
+        val pattern = Pattern.compile(URI_DETECTOR, Pattern.CASE_INSENSITIVE);
+        val matcher = pattern.matcher(url.trim());
+        return matcher.matches();
     }
 
     private val normalizedDocs =
         documents.map { (path, header) -> path.relativeTo(basePath).normalize() to header }.toMap()
 
     override fun resolveReference(source: Path, refTo: String): Reference? {
-        if (URI_DETECTOR.matches(refTo)) return null
+
+        if (refTo.startsWith(MAILTO_DETECTOR)) return null
```

Comment:
I really don't get what's the reason of doing multiple matches - how about simply slightly adjusting original regex to catch url scheme and checking that it's http, https, ftp or mailto?